CMakeLists.txt

```diff
@@ -244,6 +244,14 @@ if (EXISTS "${CMAKE_SOURCE_DIR}/.git" AND GIT)
         OUTPUT_VARIABLE TARANTOOL_GIT_VERSION
         OUTPUT_STRIP_TRAILING_WHITESPACE
         WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+    # Append '-dev' suffix to nightly builds, which don't have exact tags.
+    execute_process (COMMAND ${GIT} tag --points-at HEAD
+        OUTPUT_VARIABLE TARANTOOL_GIT_EXACT_TAG
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+    if (TARANTOOL_GIT_EXACT_TAG STREQUAL "")
+        set(TARANTOOL_GIT_VERSION "${TARANTOOL_GIT_VERSION}-dev")
```

Comment:
Can we report `2.10.0-beta1-0-g7da4b1438` as just `2.10.0-beta1` and `2.10.0-beta1-1-gfb7fe96b5` as `2.10.0-beta1-1-gfb7fe96b5-dev`?

Reply:
We can, but I don't think we should hide git SHA for tagged versions, because it's more reliable to see the SHA (suppose someone adds a tag locally without pushing it to master and builds the package, then there will be no way to figure out the commit it was built from). Why do you think it should be hidden for tagged versions?

Reply:
I don't mind output it separately from the version string, but I would prefer a 'clean' version. One of the general ideas of the new policy was to make tarantool versions visually distinguishable between dev/nightly builds and releases (pre-release in this case). So I would do everything: add the `-dev` suffix, drop the hash from (pre)releases.

BTW, will this machinery work in the same way, when we'll create packages: `OS=<...> DIST=<...> make -f .gitlab.mk package`? I guess, we should generate the version file in packpack in the same way as we do it here. That's strange that it is not the same as the `VERSION` environment variable content, just `git describe --long --always` (see [here](https://github.com/packpack/packpack/blob/0154f5e5af03126744e3ff35648de7264a011fcd/pack/tarball.mk#L16)).